simulate_from_snapshot/src/rpc_server.rs

```diff
@@ -107,33 +119,40 @@ impl TestRpcServer for RpcServerImpl {
         let account_filters = config
             .as_ref()
             .and_then(|x| x.filters.as_ref())
-            .map(|filters| filters.iter().map(AmAccountFilterType::from).collect_vec())
-            .unwrap_or_default();
-
+            .map(|filters| filters.iter().map(AmAccountFilterType::from).collect_vec());
         let commitment = Commitment::from(commitment);
 
-        let gpa = self
-            .storage
-            .get_program_accounts(program_id, Some(account_filters), commitment)
+        let (program_accounts, token_account_mints) = self
```

Comment:
the decision if token store or not should be earlier and explicitly and not rely on program_account_storage returning Err